File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        my_set = {}
        for i in range(len(nums)):
            if nums[i] in my_set and abs(i - my_set[nums[i]]) <= k:
                return True


            my_set[nums[i]] = i
        return False            

File: Data Structures & Algorithms/rotate-array/submission-2.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        vector<int> rot = nums;
        int len = nums.size();
        for (int i = 0; i < len; i++) {
            nums[(i + k) % len] = rot[i];
        }
    }
};